### DISCUSSION TITLE
Is this a good pattern for dealing with arrays of objects?

### DISCUSSION THREAD

**Discussion by GollyJer** [pmndrs/valtio#316] [Q&A] [answered]
I'm playing around with arrays of objects and nested arrays at [this codesandbox](https://codesandbox.io/s/react-valtio-array-of-objects-mhdoz?file=/src/App.tsx).

The best I could come up with is to require every object to have a key and then manipulate the arrays by object key. 
It works but is it a good approach?

**Comment by dai-shi** [comment, marked as answer]
> require every object to have a key

Yeah, I guess it's a good and predictable method, especially with complex structure.

There might be some other ways, but if we were to render an array in React, we would need stable keys anyway.

It's nice to collect more practices, so thanks for sharing!
